packages/domains-table/src/domains-table/index.tsx

```diff
@@ -16,15 +18,52 @@ interface DomainsTableProps {
 
 export function DomainsTable( { domains, fetchSiteDomains, isAllSitesView }: DomainsTableProps ) {
 	const { __ } = useI18n();
+	const [ selectedDomains, setSelectedDomains ] = useState< PartialDomainData[] >( [] );
+
+	useLayoutEffect( () => {
+		setSelectedDomains( [] );
```

Comment:
It's not clear to me why we'd want the selection to get cleared when the domain list changes. For example if the user refocuses the tab React Query might refetch the domains list. And if there are changes this will cause the selection to be wiped.
I see that one issue is object equality.
How about if we store `selectedDomains` as a `Set<string>`? That way if React Query fetches some new domains we can correctly apply the users old selection onto the new domains list?

Reply:
We have the list stored in Redux and then we're fetching it from the endpoint with React Query again. Is this list idempotent? Is it a problem if we don't clear it?

Reply:
The problem would be that checks like `selectedDomains.includes( domain )` are using object equality. If React Query fetches a new list then changes are those objects will be new objects and object equality won't treat them the same.

> Is it a problem if we don't clear it?

It's a bit of a weird case, but if the user selects a domain, and then after a fetch that domain gets removed from the table, that'd mean `areAllDomainsSelected` might get stuck as `true` position. It'd think there's one domain selected, even though that domain isn't in the table.

Reply:
Thanks for this discussion, @p-jackson.

What I'm doing now is, whenever the list changes, remove the dangling domains (i.e. selected but not in the new `domains` list) from the bulk selection set. Here's the diff: https://github.com/Automattic/wp-calypso/pull/80826/commits/12b773e6262417d15b259fe1ca008941230ac883.

Reply:
Looks great